extensions/proposals/WEBGL_video_texture/extension.xml

```diff
@@ -20,14 +20,21 @@
   <overview>
     <mirrors href="https://www.khronos.org/registry/gles/extensions/OES/OES_EGL_image_external.txt"
              name="OES_EGL_image_external">
-      <addendum>Defines a new texture target <code>TEXTURE_VIDEO_IMAGE</code>.</addendum>
+      <addendum>Defines a new texture target <code>TEXTURE_VIDEO_IMAGE_WEBGL</code> and a new sampler type <code>GL_SAMPLER_VIDOE_IMAGE_WEBGL</code></addendum>
       <addendum>Provides a mechanism for binding <code>HTMLVideoElement</code> stream to video texture targets.</addendum>
       <addendum>Provides time of frame, texture width and height of <code>HTMLVideoElement</code>'s texture binding.</addendum>
     </mirrors>
 
     <features>
       <feature>Add support for <code>WEBGL_video_texture</code>
       binding of HTMLVideoElement.</feature>
+      <feature>Bind textures to <code>TEXTURE_VIDEO_IMAGE_WEBGL</code> may unbind textures binded to <code>TEXTURE_2D</code>.</feature>
```

Comment:
Can it be more specific than "may"? I think the implementation should guarantee that these binding points are distinct, even if it has to emulate it.

Reply:
"may" here refers two different situations of current implementation. On Windows, it is emulated by a texture 2D, but on Android, the implementation should be a real distinct target.
To remove "may", there are two ways:
1. bind texture to TEXTURE_VIDEO_IMAGE_WEBGL unbind textures bound to TEXTURE_2D
This is straight for Windows platform but not necessary for Android.
2. Bind texture to TEXTURE_VIDEO_IMAGE_WEBGL doesn't unbind textures bound to TEXTURE_2D
For Android, it is true, but for Windows platform, this will have complex mechanism to find a empty texture unit (and maybe not exisit)

Because I tried method 2 first and find lots of potential issues, so I tend to use method 1. WDYT?

Reply:
I definitely think we need to spec it one way or the other.

I can see why it might be difficult to do option 2. I wasn't thinking of trying to use an empty texture unit - just use the user-provided glActiveTexture unit. The tricky part is making sure that the correct texture is bound to the texture unit's underlying binding points, each time the texture unit is used. I don't think it's ever possible to observe both binding points (TEXTURE_2D and TEXTURE_VIDEO_IMAGE_WEBGL) at the same time, but I could be wrong about that.

I would be ok specifying that it MUST replace the TEXTURE_2D binding point. However, at that point, I would prefer to say that it *IS* the TEXTURE_2D binding point. Is there any way to get rid of the extra TEXTURE_VIDEO_IMAGE_WEBGL binding point (and enum)? (Sorry, haven't had a chance to review whether this would work or not.)

Reply:
We're going to need to do some patching magic for tex unit selection (uniform1i) at draw time anyway, so that means we should be able to de-alias TEXTURE_2D[0] and TEXTURE_VIDEO_IMAGE_WEBGL[0]. I think we should treat these as always distinct, and binding to TEXTURE_VIDEO_IMAGE_WEBGL will not change bindinds to TEXTURE_2D and vice versa.

Reply:
Thanks for your reviewing!. I think I was persuaded by the uniform1i magic patching. I think the progress should be:
1. Bind TEXTURE_VIDEO_IMAGE_WEBGL won't do real binding.
2. Call sharing video frame API will unbind current TEXTURE_2D and do real binding for sharing.
3. Check sampler type in uniform1i (if possible) if the active texture unit has both TEXTURE_2D and TEXTURE_VIDEO_IMAGE_WEBGL bound and then choose the correct one bind to the unit and do real uniform1i.

WDYT?

Reply:
Off-hand, seems good to me.

Reply:
I try to update the description, PTAL if you have time.